iceoryx_hoofs/filesystem/include/iox/detail/filesystem.inl

```diff
@@ -39,15 +39,17 @@ inline bool isValidPathEntry(const iox::string<StringCapacity>& name,
         // AXIVION Next Construct AutosarC++19_03-A3.9.1: Not used as an integer but as actual character
         const char c{name[i]};
 
-        // AXIVION DISABLE STYLE AutosarC++19_03-A0.1.1, FaultDetection-UnusedAssignments : False positive, variable IS used
+        // AXIVION DISABLE STYLE FaultDetection-UnusedAssignments : False positive, variable IS used
```

Comment:
Good grief. These variables are used just below. This is a declaration of bankruptcy of Axivion.

Reply:
Could the message that you are trying to suppress have been `Result of assignment is not used along some path(s)`? In that case Axivion is quite right (for `isCapitalLetter`, `isNumber`, and `isSpecialCharacter`), because due to the shortcut semantics of C/C++ not all paths are taken in all cases, so some of the variables *are* unused along some execution paths. If you are not interested in this (not so useful) message, I suggest you rather deactivate it in the configuration than to fight it in the code.

Reply:
Let's look at line 54, because that's one that matters here: if `isSmallLetter` is `true` then `!isSmallLetter` is `false`, therefore `isCapitalLetter` is not evaluated at all because `(!isSmallLetter && !isCapitalLetter)` cannot become `true` anymore. Therefore, the initialization assignment to `isCapitalLetter` (in line 46) is indeed unused in case `isSmallLetter` is initialized to `true` (in line 45). The same argument can be made for `isNumber` and `isSpecialCharacter`. Only the assignment to `isSmallLetter` is always required because it is the first variable in line 54 and always evaluated. But for the other three the message `not used along some path(s)` is correct (but perhaps not very useful to you).

Reply:
@sbellon True. But there are evaluation paths for each variable that depend on it. Hence it should not detect it as unused. But there may be a good reason it behaves like it does.

Thanks for the detailed explanation.